File: 백준/Silver/1158. 요세푸스 문제/요세푸스 문제.cs
public class Starter
{
    public static void Main(string[] args)
    {
        string[] input = Console.ReadLine().Split();
        int n = int.Parse(input[0]);
        int k = int.Parse(input[1]);
        
        List<int> circle = new List<int>();
        for (int i = 0; i < n; i++)
        {
            circle.Add(i + 1);
        }

        int pointer = 0;
        int[] result = new int[n];
        for (int i = 0; i < n; i++)
        {
            pointer = (pointer + k - 1) % circle.Count;
            result[i] = circle[pointer];
            circle.RemoveAt(pointer);
        }

        Console.WriteLine($"<{string.Join(", ", result)}>");
    }
}

File: 백준/Silver/1158. 요세푸스 문제/요세푸스 문제.cc
#include <iostream>

int main(int argc, char *argv[]) {
    int n, k;
    std::cin >> n >> k;

    int circle[n];
    for (int i = 0; i < n; ++i) {
        circle[i] = i + 1;
    }

    int pointer = n - 1;
    int result[n];
    for (int i = 0; i < n; ++i) {
        for (int j = 0; j < k; ++j) {
            do {
                pointer = (pointer + 1) % n;
            } while (circle[pointer] == 0);
        }
        result[i] = circle[pointer];
        circle[pointer] = 0;
    }

    std::cout << "<";
    for (int i = 0; i < n-1; ++i) {
        std::cout << result[i] << ", ";
    }
    std::cout << result[n-1] << ">";
    return 0;
}